run.c

```diff
@@ -132,11 +132,11 @@ void checkpoint_init_weights(TransformerWeights *w, Config* p, float* f, int sha
     w->rms_ffn_weight = ptr;
     ptr += p->n_layers * p->dim;
     w->w1 = ptr;
-    ptr += p->n_layers * p->dim * p->hidden_dim;
+    ptr += (unsigned int)p->n_layers * p->dim * p->hidden_dim;
```

Comment:
this and below should really be unsigned long long, or maybe uint64_t. Possibly long long or int64_t,